src/wallet/rpcwallet.cpp

```diff
@@ -558,6 +559,104 @@ UniValue signmessage(const UniValue &params, bool fHelp)
     return EncodeBase64(&vchSig[0], vchSig.size());
 }
 
+UniValue signdata(const UniValue &params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() < 3 || params.size() > 4)
+        throw runtime_error(
+            "signdata \"bitcoinaddress\" \"msgFormat\" \"message\"\n"
+            "\nSign message for use with the CHECKDATASIG instruction."
+            "\nAs per the CHECKDATASIG operation, this RPC normally signs the SHA256 of the provided message\n"
+            "\nunless the 'hash' message format is specified.\n" +
+            HelpRequiringPassphrase() +
+            "\n"
+            "\nArguments:\n"
+            "1. \"bitcoinaddress\"  (string, required) The bitcoin address to use for the private key.\n"
+            "2. \"msgFormat\"       (string, required) Use \"string\", \"hex\", or \"hash\" to specify the message "
+            "encoding.\n"
```

Comment:
I wasn't clear that this "hash" was a 256bit hash only and tried shorter hashes which were rejected.  Maybe a comment there to clarify it to the user?

Reply:
fixed